src/Serializer.php

```diff
@@ -136,4 +165,16 @@ public static function wrapClosures(&$data, SerializerInterface $serializer)
             }
         }
     }
+
+    /**
+     * Calculates a signature for a closure's serialized data.
+     *
+     * @param string $data Serialized closure data.
+     *
+     * @return string Signature of the closure's data.
+     */
+    private function calculateSignature($data)
+    {
+        return hash_hmac('sha256', $data, $this->signingKey);
```

Comment:
consider returning the raw value (passing `true` to the fourth parameter), using that for comparisons, then encoding using `base64_encode/decode`. It's more compact (88 bytes vs 128) and comparing raw strings is typically better...